src/mountinfo/mountinfo.c

```diff
@@ -356,6 +385,174 @@ find_mounts(struct args *args)
+			 */
+			for (i = n = 0; i < 3; ++i) {
+				if (*r >= '0' && *r <= '7') {
+					n <<= 3;
+					n |= *r++ - '0';
+				} else {
+					break;
+				}
+			}
+			if (n)
+				*w++ = n;
+		}
+	}
+	*w = '\0';
+	return beg;
+}
+
+static int check_is_mounted(RC_STRINGLIST *list RC_UNUSED, struct args *args,
+    char *from RC_UNUSED, char *to, char *fstype RC_UNUSED,
+    char *options RC_UNUSED, int netdev RC_UNUSED)
+{
+	return strcmp(args->check_mntpath, unescape_octal(to)) == 0 ? 0 : -1;
+}
+
+static int is_mounted(const char *mntpath)
+{
+	size_t num_mounts = 0;
+	struct args args = { .process = check_is_mounted, .check_mntpath = mntpath };
+	RC_STRINGLIST *l = find_mounts(&args, &num_mounts);
+	rc_stringlist_free(l);
+	return num_mounts > 0;
+}
+
+static pid_t run_umount(const char *mntpath,
+	const char **umount_args, int umount_args_num)
+{
+	struct exec_args args;
+	struct exec_result res;
+	const char *argv[UMOUNT_ARGS_MAX + 3];
+	int k, i = 0;
+
+	argv[i++] = "umount";
+	for (k = 0; k < umount_args_num; ++k)
+		argv[i++] = umount_args[k];
+	argv[i++] = mntpath;
+	argv[i++] = NULL;
+
+	args = exec_init(argv);
+	args.redirect_stdout = args.redirect_stderr = EXEC_DEVNULL;
+	res = do_exec(&args);
+	if (res.pid < 0)
+		eerrorx("%s: failed to run umount: %s", applet, strerror(errno));
+	return res.pid;
+}
+
+static void fuser_run(struct run_queue *rp, const char *fuser_opt)
+{
+	static int fuser_exec_failed = 0;
+	const char *argv[] = { "fuser", fuser_opt, rp->mntpath, NULL };
+	struct exec_result res;
+	struct exec_args args;
+
+	/* if exec failed, fuser likely doesn't exist. so don't retry */
+	if (fuser_exec_failed)
+		return;
+
+	args = exec_init(argv);
+	args.redirect_stdout = EXEC_MKPIPE;
+	args.redirect_stderr = EXEC_DEVNULL;
+	res = do_exec(&args);
+	if (res.pid < 0) {
+		fuser_exec_failed = 1;
+	} else {
+		rp->fuser_pid = res.pid;
+		rp->fuser_stdoutfd = res.proc_stdout;
+	}
+}
+
+static int fuser_decide(struct run_queue *rp,
+	const char *fuser_opt, const char *fuser_kill_prefix)
+{
+	char buf[1<<12];
+	char selfpid[64];
+	ssize_t n;
+
+	if (rp->fuser_stdoutfd < 0)
+		return 0;
+
+	buf[0] = ' ';
+	n = read(rp->fuser_stdoutfd, buf + 1, sizeof buf - 3);
+	close(rp->fuser_stdoutfd);
+	rp->fuser_stdoutfd = -1;
+	if (n < 0)
+		return 0;
+	buf[n+1] = ' ';
+	buf[n+2] = '\0';
+	snprintf(selfpid, sizeof selfpid, " %lld ", (long long)getpid());
+
+	if (strstr(buf, selfpid)) {
+		/* lets not kill ourselves */
+		eerror("Unmounting %s failed because we are using it", rp->mntpath);
+		return -1;
+	} else if (strcmp(buf, "  ") == 0) {
+		if (rp->try_count >= TRY_MAX) {
+			eerror("Unmounting %s failed but fuser finds no one using it", rp->mntpath);
+			return -1;
+		}
+		/* it's possible that whatever was using the mount stopped
+		 * using it now, so allow 1 more retry */
+		rp->try_count = TRY_MAX;
+		return 0;
+	} else if (strcmp(buf, " - ") == 0) {
+		/* NRK: huh?? */
+		eerror("Unmounting %s failed", rp->mntpath);
+		return -1;
```

Comment:
I don't quite get when this is supposed to happen. Git blame takes me [to here](https://github.com/OpenRC/openrc/commit/ff162b39ace915ed9803ff7d091157855cf6d48c) but why would fuser not being preset output "-"?

Reply:
> Git blame takes me [to here](https://github.com/OpenRC/openrc/commit/ff162b39ace915ed9803ff7d091157855cf6d48c) but why would fuser not being preset output "-"?

Was pointed out by @dwfreed that the pid var is set to '-' initially and the command is only run if fuser is preset. I missed this completely.

Removed this part since we keep track of exec failure and treat it as a proxy for fuser not being present.